### PR TITLE
Improved UnorderedCSVDiffer speed

### PR DESCRIPTION
**Pull request by PaulTalbot-INL**
Marginally increases the speed of the UnorderedCSVDiffer (5-10% faster).

Also adds external model test as prereq for external model reseed test, which checks the same files and sometimes caused a missing CSV error.

### CODE REVIEW

File: tests/framework/tests [hunk 1]
```diff
@@ -209,6 +209,7 @@
   type = 'RavenFramework'
   input = 'test_external_reseed.xml'
   output = 'externalModel/testPrintHistorySet_dump_0.csv externalModel/testPrintHistorySet_dump_1.csv externalModel/testPrintHistorySet_dump_2.csv'
+  prereq = testExternalModel
```
Comment (alfoa) [inline tests/framework/tests]: This modification is already in dataobject-rework ...why no conflict is shown??? O_o

Comment (PaulTalbot-INL) [reply tests/framework/tests]: If the change is identical, I think maybe it can handle coinciding both effects.

Comment (PaulTalbot-INL) [reply tests/framework/tests]: Let me rebase to assuage those concerns.

**Comment by PaulTalbot-INL** [comment]
That didn't work as desired.   A moment...

**Pull request merged**